lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -11,6 +11,7 @@ def initialize(input, suspended_action:)
         @playbook = input.values.first[:input][:action_input][:script]
         @root = working_dir
         @verbosity_level = input.values.first[:input][:action_input][:verbosity_level]
+        @check_mode = input.values.first[:input][:action_input][:ansible_inventory][:all][:vars][:check_mode]
```

Comment:
Just out of curiosity, why is check mode nested so deep when compared with the playbook script?

Reply:
Because the script is a general action_input that is defined in the remote_execution project: specific in foreman_remote_execution/app/lib/actions/remote_execution/run_host_job.rb